### ISSUE TITLE
global usings when using asp.net core APIs in a class library

### ISSUE THREAD

**Issue by freddyrios** [dotnet/AspNetCore.Docs#28380]
[EDIT by guardrex to add the metadata. @freddyrios ... In the future, open an issue with the **This page** feedback form at the bottom of any article to automatically add the metadata and ping the author of the topic.]

This issue https://github.com/dotnet/AspNetCore.Docs/issues/22048 links to [documentation](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/target-aspnetcore?view=aspnetcore-7.0&tabs=visual-studio#use-the-aspnet-core-shared-framework) on how to reference the core shared framework.

However, when doing so you get no implicit usings like you do on a project that uses Microsoft.NET.Sdk.Web. Is there any documented way to enable that for class libraries?

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 98629146-7d64-9f48-6e45-51bafe8c7116
* Version Independent ID: cdb2b89f-bf1c-f313-0062-7f6b7ef8b611
* Content: [Use ASP.NET Core APIs in a class library](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/target-aspnetcore?view=aspnetcore-7.0&tabs=visual-studio#use-the-aspnet-core-shared-framework)
* Content Source: [aspnetcore/fundamentals/target-aspnetcore.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/fundamentals/target-aspnetcore.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **scaddie**

**Comment by freddyrios** [comment]
Context: while adding a web subsystem to a class library used by some hosts, I tried using the minimal APIs. 

It works nicely, but had the same trouble as the issue I linked and after fixing the framework reference, it still seemed to fail but was just the missing (global) using. 

Since the doc I linked seems to be precisely about running asp.net code in a class library *and* that's about the only case where you won't get an intellisense suggestion for the missing namespace it seemed like that right place for it.

**Comment by guardrex** [comment]
> It works nicely, but had the same trouble as the issue I linked and after fixing the framework reference, it still seemed to fail but was just the missing (global) using.

I don't think I understood your ask then. Can you be more specific? What exactly did you do ... show the `Using`(s) that you added to the class lib's project file that the section isn't covering.

**Comment by guardrex** [comment]
@Rick-Anderson ... Turns out that this will be one of yours to triage.

**Comment by freddyrios** [comment]
Specifically:

1. added code to a class library like the first example here https://learn.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis?view=aspnetcore-7.0. This fails to build as expected.
2. not knowing how minimal APIs worked underneath, I looked for a solution to run them in a class library and ended up finding this issue https://github.com/dotnet/AspNetCore.Docs/issues/22048 and applied the framework reference mentioned at [documentation](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/target-aspnetcore?view=aspnetcore-7.0&tabs=visual-studio#use-the-aspnet-core-shared-framework)
3. tried building again and failed. This surprised me because the failure looked the same as in step 1.
4. I tried the Sdk.Web and that worked. This lead me to understand that this was adding global usings
5. Moved back to Sdk and used intellisense to add the using at the file level for me

The end result at 5 is fine, but going from 3 to 5 was not obvious given the sequence of steps that got me there. I only tried 4 because I saw it in the issue I linked (where it is mentioned that has issues, so not proper), but without it and only the doc it could have taken much longer to figure out.
